src/python/ddapp/tabledemo.py

```diff
@@ -1194,6 +1196,14 @@ def onPropertyChanged(self, propertySet, propertyName):
         elif propertyName == 'Scene':
             self.tableDemo.sceneID = self.params.getProperty('Scene')
 
+    def pickupMoreObjects(self):
+        if len(self.tableDemo.clusterObjects) > 0: # There is still sth on the table, let's do it again!
+            print "There are more objects on the table, going at it again!"
+            self.taskTree.selectTaskByName('reach')
```

Comment:
I think your looping strategy is good.  The only issue is that it has to know the name of the starting task, 'reach' in this case, but I think that's fine.